platform/android/src/style/sources/sources.cpp

```diff
@@ -43,11 +45,16 @@ namespace android {
 jni::jobject* createJavaSourcePeer(jni::JNIEnv& env, AndroidRendererFrontend& frontend, mbgl::style::Source& coreSource) {
     std::unique_ptr<Source> peerSource = std::unique_ptr<Source>(initializeSourcePeer(coreSource));
     peerSource->setRendererFrontend(frontend);
-    jni::jobject* result = peerSource->createJavaPeer(env);
+    jni::jobject* result = peerSource->getJavaPeer(env);
     peerSource.release();
     return result;
 }
 
+jni::jobject* removeSourceFromMap(std::unique_ptr<mbgl::style::Source>&& coreSource) {
```

Comment:
This method is confusing. Why is it named `removeSourceFromMap`, when it doesn't do that? Why does it construct a unique pointer and then immediate release it?

Reply:
`unique_ptr` usage is not necessary and can be removed from both this function and `createJavaSourcePeer`.